### PR TITLE
Fix phpstan issue with tests

### PR DESCRIPTION
**Pull request by VincentLanglet**
This provide a way to fix phpstan error with filters in test directory, because `query` and `queryParameters` doesn't exist on both `ProxyQuery` and `QueryBuilder`.

### CODE REVIEW

File: tests/Filter/CallbackFilterTest.php [hunk 1]
```diff
@@ -81,33 +81,33 @@ public function testFilterException(): void
     {
         $this->expectException(\RuntimeException::class);
```
Comment (phansys) [inline tests/Filter/CallbackFilterTest.php:82]: I know this is not part of the issue fixed in this PR, but could you please move this assertion just before the line intended to throw the exception?

File: tests/Filter/FilterTestCase.php [hunk 2]
```diff
@@ -18,17 +18,44 @@
...
+    {
+        $queryBuilder = $proxyQuery->getQueryBuilder();
+        if (!$queryBuilder instanceof TestQueryBuilder) {
+            throw new \InvalidArgumentException('The query builder should be build with "createQueryBuilderStub()".');
+        }
+
+        $this->assertSame($expected, $queryBuilder->queryParameters);
+    }
+
+    public function createQueryBuilderStub(): TestQueryBuilder
```
Comment (phansys) [inline tests/Filter/FilterTestCase.php:54]:
```suggestion
    final protected function createQueryBuilderStub(): TestQueryBuilder
```

File: tests/Filter/FilterTestCase.php [hunk 3]
```diff
@@ -18,17 +18,44 @@
 use Doctrine\ORM\Query\Expr\Orx;
 use Doctrine\ORM\QueryBuilder;
 use PHPUnit\Framework\TestCase;
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
+
+class TestQueryBuilder extends QueryBuilder
```
Comment (phansys) [inline tests/Filter/FilterTestCase.php:23]:
```suggestion
abstract class TestQueryBuilder extends QueryBuilder
```

File: tests/Filter/FilterTestCase.php [hunk 4]
```diff
@@ -18,17 +18,44 @@
...
+{
+    /** @var string[] */
+    public $query = [];
+
+    /** @var string[] */
+    public $queryParameters = [];
+}
 
 class FilterTestCase extends TestCase
```
Comment (phansys) [inline tests/Filter/FilterTestCase.php:32]:
```suggestion
abstract class FilterTestCase extends TestCase
```

File: tests/Filter/FilterTestCase.php [hunk 5]
```diff
@@ -18,17 +18,44 @@
...
+        if (!$queryBuilder instanceof TestQueryBuilder) {
+            throw new \InvalidArgumentException('The query builder should be build with "createQueryBuilderStub()".');
+        }
+
+        $this->assertSame($expected, $queryBuilder->queryParameters);
+    }
+
+    public function createQueryBuilderStub(): TestQueryBuilder
+    {
+        $testCase = $this;
```
Comment (phansys) [inline tests/Filter/FilterTestCase.php:57]: We can use `$this`.
```suggestion
```

Comment (VincentLanglet) [reply tests/Filter/FilterTestCase.php]: 1) Sonata\DoctrineORMAdminBundle\Tests\Filter\BooleanFilterTest::testFilterArray
Error: Using $this when not in object context

File: tests/Filter/FilterTestCase.php [hunk 6]
```diff
@@ -18,17 +18,43 @@
 use Doctrine\ORM\Query\Expr\Orx;
 use Doctrine\ORM\QueryBuilder;
 use PHPUnit\Framework\TestCase;
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
 
-class FilterTestCase extends TestCase
+class TestQueryBuilder extends QueryBuilder
```
Comment (phansys) [inline tests/Filter/FilterTestCase.php:23]: IMO, we should declare this fixture after the PSR-4 class.

Comment (VincentLanglet) [reply tests/Filter/FilterTestCase.php]: Done

File: tests/Filter/FilterTestCase.php (hunk 5 above)
Comment (phansys) [reply tests/Filter/FilterTestCase.php]: Did you remove the `static` keyword in the lambda where this variable is used?

**Review by phansys — DISMISSED** [review]

File: tests/Filter/FilterTestCase.php (hunk 5 above)
Comment (phansys) [reply tests/Filter/FilterTestCase.php]: I'd like to use the changes in the tests as base for the checks I need to add at #1358.
@VincentLanglet, could you please confirm if this variable can be removed?

**Review by phansys — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thank you @VincentLanglet

**Pull request merged**